server/api.ts

```diff
@@ -124,7 +132,10 @@ const parseWarmup = async (warmup: string, mod: WarmupMod, tourney: string) => {
     throw new Error("Warmup map too long (max 3 minutes)");
   }
 
-  return `https://osu.ppy.sh/beatmapsets/${mapData.beatmapSetId}#taiko/${warmupMapId}`;
+  return `https://osu.ppy.sh/beatmapsets/${mapData.beatmapSetId}#${
+    // GameMode defines catch but in beatmap url it's fruits
+    tourneyData.mode == "catch" ? "fruits" : tourneyData.mode
```

Comment:
hardcoding this with only 2 options is shooting ourselves in the foot (again)

Reply:
This should work for all game modes (as long as `tourneyData.mode` is defined properly for `osu`/`mania` if these game modes exists in future tournies). It only converts `catch` to `fruits` and passes everything else in as is.